src/test/java/io/pravega/connectors/flink/FlinkTableITCase.java

```diff
@@ -211,6 +226,243 @@ public void testBatchTable() throws Exception {
+     * write and read from Pravega stream using {@link StreamTableEnvironment}
+     * @throws Exception
+     */
+    @Test
+    public void testStreamingTableUsingDescriptor() throws Exception {
+
+        final String scope = setupUtils.getScope();
+        final String streamName = "stream";
+        Stream stream = Stream.of(scope, streamName);
+        this.setupUtils.createTestStream(stream.getStreamName(), 1);
+
+        StreamExecutionEnvironment env = StreamExecutionEnvironment.createLocalEnvironment().setParallelism(1);
+        StreamTableEnvironment tableEnv = TableEnvironment.getTableEnvironment(env);
+
+        PravegaConfig pravegaConfig = PravegaConfig.fromDefaults()
+                .withControllerURI(setupUtils.getControllerUri())
+                .withDefaultScope(setupUtils.getScope());
+
+        Pravega pravega = new Pravega();
+        pravega.tableSinkWriterBuilder()
+                .withRoutingKeyField("category")
+                .forStream(stream)
+                .withPravegaConfig(pravegaConfig);
+        pravega.tableSourceReaderBuilder()
+                .withReaderGroupScope(stream.getScope())
+                .forStream(stream)
+                .withPravegaConfig(pravegaConfig);
+
+        StreamTableDescriptor desc = tableEnv.connect(pravega)
+                .withFormat(
+                    new Json()
+                            .failOnMissingField(false)
+                            .deriveSchema()
+                )
+                .withSchema(new Schema().field("category", Types.STRING()).field("value", Types.INT()))
+                .inAppendMode();
+        desc.registerTableSourceAndSink("test");
+
+        final Map<String, String> propertiesMap = DescriptorProperties.toJavaMap(desc);
+        final TableSink<?> sink = TableFactoryService.find(StreamTableSinkFactory.class, propertiesMap)
+                .createStreamTableSink(propertiesMap);
+        final TableSource<?> source = TableFactoryService.find(StreamTableSourceFactory.class, propertiesMap)
+                .createStreamTableSource(propertiesMap);
+
+        Table table = tableEnv.fromDataStream(env.fromCollection(SAMPLES));
+        table.writeToSink(sink);
+
+        tableEnv.registerTableSource("samples", source);
+
+        // select some sample data from the Pravega-backed table, as a view
+        Table view = tableEnv.sqlQuery("SELECT * FROM samples WHERE category IN ('A','B')");
+
+        // write the view to a test sink that verifies the data for test purposes
+        tableEnv.toAppendStream(view, SampleRecord.class).addSink(new TestSink(SAMPLES));
+
+        // execute the topology
+        try {
+            env.execute();
+            Assert.fail("expected an exception");
+        } catch (JobExecutionException e) {
+            // we expect the job to fail because the test sink throws a deliberate exception.
+            Assert.assertTrue(e.getCause() instanceof TestCompletionException);
```

Comment:
I'm not clear on why you want it to throw in this test case. Also, it seems to be throwing a generic exception, is there a more precise behavior you want to capture?

Reply:
Table factory uses java service loader to load all the available factories. We have defined all the table factory implementations that we support in this file.